agent/checker.go

```diff
@@ -35,6 +37,14 @@ type CheckResult struct {
 	controls *check.Controls
 }
 
+func (cr *CheckResult) PrettyPrint(f *os.File) {
```

Comment:
Consider using a `StringWriter` interface instead of `*os.File` as a type here. In general, interfaces rather than concrete implementations make the code more pluggable and easier to test.